server/test/api/transferts.js

```diff
@@ -0,0 +1,150 @@
+*/
+describe('The /transferts API endpoint :: ', function () {
+  'use strict';
+
+  var agent = chai.request.agent(helpers.baseUrl);
+
+  /** login before each request */
+  beforeEach(helpers.login(agent));
+
+  /** Test with the current date */
+  var date = new Date();
+
+  /** Primary Cash Object */
+  var primaryCash = {
+      uuid        : uuid(),
+      project_id  : 1,
+      type        : 'E',
+      date        : date,
+      currency_id : 2,
+      account_id  : 3627,
+      cost        : 777,
+      description : 'Test Caisse Transfert/' + date.toString(),
+      cash_box_id : 2,
+      origin_id   : 1,
+      user_id     : 1
+  };
+
+  var wrongPrimaryCash = {
+    uuid        : uuid(),
+    project_id  : 1,
+    type        : 'X',
+    date        : date,
+    currency_id : 2,
+    account_id  : 3627,
+    cost        : 555
+  };
+
+  /** Expected keys */
+  var expectedKeys = Object.keys(primaryCash);
+  expectedKeys.push('document_uuid');
+
+  it('POST /transferts should create a new transfert records in primary_cash, primary_cash_item and journal', function () {
+    return agent.post('/transferts')
+      .send(primaryCash)
+      .then(function (res) {
+        expect(res).to.have.status(201);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /transferts should return a list of all transferts', function () {
+    return agent.get('/transferts')
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res.body).to.not.be.empty;
+        expect(res.body[0]).to.have.all.keys(expectedKeys);
+
+        /**
+        * @fixme: The `date` column in primary cash must be in `datetime` format
+        * for getting the correct last date in time precision instead of getting a rondom date
+        * without time distinction
+        *
+        * The first record in the returned data must be the last inserted record
+        * due to `ORDER BY {date} DESC` clause in MySQL
+        */
+        expect(res.body[0].uuid).to.exist;
+        expect(res.body[0].uuid).to.be.equals(primaryCash.uuid);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /transferts/?limit=1 should return a limited set of results', function () {
+    return agent.get('/transferts/?limit=1')
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res.body).to.not.be.empty;
+        expect(res.body).to.have.length(1);
+        expect(res.body[0]).to.have.all.keys(expectedKeys);
+      })
+      .catch(helpers.handler);
+  });
+
+  it('GET /transferts/{uuid} should return specific transfert record according a {uuid} given', function () {
+    return agent.get('/transferts/' + primaryCash.uuid)
+      .then(function (res) {
+        var result = res.body;
+        delete result.document_uuid;
+
+        expect(res).to.have.status(200);
+        expect(result).to.not.be.empty;
+
+        /**
+        * @fixme: we must be able to compare date and time for a need of accuracy
```

Comment:
It seems to me a useful helper to have would be a "compare properties" function in the helpers library that could determine the type of the value (date, number, string, etc), and automatically convert values to dates to compare.

It would certainly make these tests easier.  I imagine something that looks like this:
```js
/**
* Compares the {keys} on both object A and object B, making sure that
*  1. They exist
*  2. They are equal
* @param {Object} objectA A JSON object
* @param {Object} objectB A JSON object
* @param {Array} keys A list of keys to compare between objectA and objectB
* @returns {boolean} assertion A boolean indicating that objectA's key/values are equal to ObjectB's key/values. 
* @example
* var a = { id: 1, name: 'mbayopanda', year : 2015 };
* var b = { id: 1, name: 'mbayopanda', year: 2016 };
* expect(helpers.equalProperties(a, b, [ 'id', 'name' ])).to.be.true;
* expect(helpers.equalProperties(a, b, [ 'id', 'name', 'year' ]).to.be.false;
*/
helpers.equalProperties(objectA, objectB, keys);